test/functional/cmdLineTests/cmdLineTest_J9tests/src/OnOutOfMemoryErrorTest.java

```diff
@@ -0,0 +1,33 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+/**
+ * @file OnOutOfMemoryErrorTest.java
+ * @brief Generates a java.lang.OutOfMemoryError to test the command line option
+ *        -XX:OnOutOfMemoryError=
+ */
+
+class OnOutOfMemoryErrorTest {
+    public static void main(String[] args) {
+        Object test = new double[Integer.MAX_VALUE];
```

Comment:
I think this needs to be improved: with a large enough heap (>32GB) this might not throw OutOfMemoryError.
Either the command-line arguments must limit the heap to something smaller or this should allocate large objects in a loop (putting them in some collection).

Reply:
Would adding `-Xmx8G` be the correct command-line option to limit the heap to 8G?

Reply:
That should do it, but the heap doesn't need to be even that big - 1GB is probably overkill.